.github/workflows/beam_PreCommit_Spotless.yml

```diff
@@ -0,0 +1,104 @@
+#    Licensed to the Apache Software Foundation (ASF) under one
+#    or more contributor license agreements.  See the NOTICE file
+#    distributed with this work for additional information
+#    regarding copyright ownership.  The ASF licenses this file
+#    to you under the Apache License, Version 2.0 (the
+#    "License"); you may not use this file except in compliance
+#    with the License.  You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+name: PreCommit Spotless
+
+on:
+  push:
+    tags: ['v*']
+    branches: ['master', 'release-*']
+    paths:
+      - 'buildSrc/**'
+      - 'sdks/java/**'
+      - 'runners/**'
+      - 'examples/java/**'
+      - 'examples/kotlin/**'
+      - '.test-infra/jenkins/'
+      - '.github/workflows/beam_PreCommit_Spotless.yml'
+  pull_request_target:
+    branches: ['master', 'release-*']
+    paths:
+      - 'buildSrc/**'
+      - 'sdks/java/**'
+      - 'runners/**'
+      - 'examples/java/**'
+      - 'examples/kotlin/**'
+      - '.test-infra/jenkins/'
+  issue_comment:
+    types: [created]
+  schedule:
+    - cron: '0 */6 * * *'
+  workflow_dispatch:
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.sha || github.head_ref || github.ref }}-${{ github.event.sender.login }}-${{ github.event.schedule }}'
+  cancel-in-progress: true
+
+#Setting explicit permissions for the action to avoid the default permissions which are `write-all` in case of pull_request_target event
+permissions:
+  actions: write
+  pull-requests: read
+  checks: read
+  contents: read
+  deployments: read
+  id-token: none
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+jobs:
+  beam_PreCommit_Spotless:
+    name: beam_PreCommit_Spotless (Run Spotless PreCommit)
+    timeout-minutes: 120
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    if: |
+      github.event_name == 'push' ||
+      github.event_name == 'pull_request_target' ||
+      github.event_name == 'schedule' ||
+      github.event.comment.body == 'Run Spotless PreCommit'
+    steps:
+      - name: Git checkout
```

Comment:
We now consolidated these first 2 steps into `./.github/actions/setup-action`:

e.g.

https://github.com/apache/beam/blob/7e365d78c797c4cd8b933857c780b5d681e597fd/.github/workflows/beam_PreCommit_Java_Examples_Dataflow_Java11.yml#L81